src/mscorlib/shared/System/IO/Path.cs

```diff
@@ -383,11 +383,68 @@ public static string Combine(params string[] paths)
             return StringBuilderCache.GetStringAndRelease(finalPath);
         }
 
-        /// <summary>
-        /// Combines two paths. Does no validation of paths, only concatenates the paths
-        /// and places a directory separator between them if needed.
-        /// </summary>
-        private static string CombineNoChecks(ReadOnlySpan<char> first, ReadOnlySpan<char> second)
+        // Unlike Combine(), Join() methods do not consider rooting. They simply combine paths, ensuring that there
+        // is a directory separator between them.
+
+        public static string Join(ReadOnlySpan<char> path1, ReadOnlySpan<char> path2)
+        {
+            if (path1.Length == 0)
+                return path2.Length == 0 ? string.Empty : new string(path2);
```

Comment:
Why do we have the length == 0 check here and not for path2.Length in the second if?

The string ctor already does this check, so we shouldn't need it in either case:
https://github.com/dotnet/coreclr/blob/master/src/mscorlib/src/System/String.cs#L665

Reply:
Thanks, didn't know the optimization was there.